tiatoolbox/wsicore/wsireader.py

```diff
@@ -1415,39 +1621,135 @@ def read_bounds(
-        )
+        if location_is_at_requested:
+            # do this to enforce output size is as defined by input bounds
+            im_region = utils.transforms.imresize(
+                img=im_region, output_size=size_at_requested
+            )
+        else:
+            im_region = utils.transforms.imresize(
+                img=im_region,
+                scale_factor=post_read_scale,
+                output_size=size_at_requested,
+            )
 
         if self.mode == "rgb":
             im_region = utils.transforms.background_composite(image=im_region)
         return im_region
 
+    def attach_to_reader(self, ref_reader_info: WSIMeta):
+        """Change self metadata to synchronize read with `reader_info`.
+
+        Modify the metadata internally to allow reading correct scaling with
+        respect to the resolution units of the source `ref_reader_info`. This is
+        applicable only for resolutions of `mpp`, `power`, and `baseline` and
+        usable for `read_bounds`.
+
+        Args:
+            ref_reader_info (WSIMeta): reference metadata to allow synchronize read
+                when using the reference input information for `read_bounds`.
+                Check the example to see how this will operate.
+
+        Examples:
+            >>> wsi_reader = WSIReader(wsi_path)
+            >>> vrt_reader = VirtuReader(tile) # at 14mpp for example
+            >>> vrt_reader.attach_to_reader(wsi_reader.info)
+            >>> bound_in_wsi = np.array([500, 600, 500, 600])
+            >>> roi_img = wsi_reader.read_bound(
+            ...     bound_in_wsi, resolutions=0.25, units='mpp')
+            >>> roi_msk = vrt_reader.read_bound(
+            ...     bound_in_wsi, resolutions=0.25, units='mpp')
+            >>> # img and msk at same resolution, with same size, expected same content
+            >>> assert roi_img.shape[0] == roi_msk.shape[0]
+            >>> assert roi_img.shape[1] == roi_msk.shape[1]
+            >>> assert roi_img.shape[0] == 100
+            >>> assert roi_img.shape[1] == 100
+
+        """
+        self.is_attach = True  # do we need a func to reset the info ?
+        # may be should create a sync reader class instead....
+        # to WH to match with WSI
+        if ref_reader_info.mpp is None:
+            raise ValueError("Reference `mpp` must not be None.")
+        if ref_reader_info.objective_power is None:
+            raise ValueError("Reference `objective_power` must not be None.")
+        mask_shape = np.array(self.img.shape[:2])[::-1]
+        mask_scale = ref_reader_info.slide_dimensions / mask_shape
+        mask_mpp = ref_reader_info.mpp * mask_scale
+        mask_obj = ref_reader_info.objective_power / mask_scale[0]
+        self._ref_info = copy.deepcopy(ref_reader_info)
+        self.info = WSIMeta(
+            file_path=self.info.file_path,
+            slide_dimensions=mask_shape,
+            level_count=1,
```

Comment:
Why we are keeping only one level_count (level_dimensions, level_downsaples) here? I guess we can use all the levels information in the VirtualSlideReader here.

Reply:
I'm not sure what do you mean here. VRReader always have only 1 level because it is not pyramidal (only baseline resolution). `attach_reader` just modifies its metadata such that internal reading can scale wrt the reference reader, which actually contain many levels.